server/hydrogen-render/1-render-hydrogen-to-string.js

```diff
@@ -85,60 +117,58 @@ async function renderHydrogenToString(renderOptions) {
       });
 
       child.on('close', (exitCode) => {
+        childExitCode = exitCode;
         // Exited successfully
         if (exitCode === 0) {
           resolve(data);
         } else {
-          let extraErrorsMessage = '';
-          if (childErrors.length > 1) {
-            extraErrorsMessage = ` (somehow we saw ${
-              childErrors.length
-            } errors but we really always expect 1 error)\n${childErrors
-              .map((childError, index) => `${index}. ${childError.stack}`)
-              .join('\n')}`;
-          }
-
-          let childErrorToDisplay = new Error('No child errors');
-          if (childErrors.length === 1) {
-            childErrorToDisplay = childErrors[0];
-          } else if (childErrors.length > 1) {
-            childErrorToDisplay = new Error('Multiple child errors listed above ^');
-          }
-
-          const error = new RethrownError(
-            `Child process failed with exit code ${exitCode}${extraErrorsMessage}`,
-            childErrorToDisplay
+          const childErrorSummary = assembleErrorAfterChildExitsWithErrors(
+            childExitCode,
+            childErrors
           );
-          reject(error);
+          reject(childErrorSummary);
         }
       });
 
       // When a problem occurs when spawning the process or gets aborted
       child.on('error', (err) => {
         if (err.name === 'AbortError') {
-          throw new RethrownError(
-            `Timed out while rendering Hydrogen to string so we aborted the child process after ${RENDER_TIMEOUT}ms`,
-            err
+          const childErrorSummary = assembleErrorAfterChildExitsWithErrors(
+            childExitCode,
+            childErrors
+          );
+          reject(
+            new RethrownError(
```

Comment:
Instead of throwing this error, we now reject with it which seems a lot more proper.

I'm not sure how it was able to bubble up to our try/catch before when this is in a callback 🤔 (it does do that though).